### DISCUSSION TITLE
The images were not uploaded in the order selected during the upload, but were sorted chaotically

### DISCUSSION THREAD

**Discussion by CookSleep** [open-webui/open-webui#5125] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.16
- **Operating System:** Ubuntu 22.04
- **Browser (if applicable):** Edge 128.0.2739.54

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When I select multiple images in order in the file explorer on the computer or the file manager on the phone, the images should be uploaded in the order selected.

## Actual Behavior:

The images are completely out of order.

## Description

**Bug Summary:**
The images were not uploaded in the order selected during the upload, but were sorted chaotically.

## Reproduction Details

**Steps to Reproduce:**
1. Select multiple images in order in the file explorer on the computer or the file manager on the phone
2. Confirm upload
3. Images are uploaded out of order

## Additional Information

If the images cannot be uploaded in order, LLM will most likely misunderstand the content, making the conversation extremely difficult. And manually uploading images one by one is very slow and tiring...

**Comment by tjbck** [comment]
Unable to reproduce, if anyone manages to reproduce this issue in the future, feel free to make a PR!

**Reply by CookSleep** [reply]
Well... it seems that I was tricked by the photo picker in the Android system made by Google. Using it to select multiple pictures will not upload them in order...
I'll check it again on the computer

**Reply by CookSleep** [reply]
Uploading pictures using the photo app on the Android system does not have this problem, but unfortunately, Google seems to be continuously promoting this extremely difficult-to-use photo picker :(

**Reply by CookSleep** [reply]
Strange, the problem hasn't occurred on the computer anymore.

So, this issue should only occur when using the Android photo picker on mobile, but I feel it can't be fixed. It seems to be caused by Google's confusing and poor design.
